discord/context_managers.py

```diff
@@ -64,7 +64,7 @@ async def do_typing(self) -> None:
             await asyncio.sleep(5)
 
     def __enter__(self: TypingT) -> TypingT:
-        self.task: asyncio.Task = self.loop.create_task(self.do_typing())
+        self.task: asyncio.Future = self.loop.create_task(self.do_typing())
```

Comment:
I feel like Task is more correct

Reply:
Agreed, as per [Python docs](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.create_task) `Task` is appropriate.

Reply:
ah yeah, Task is Future-like but better to change `context_manages._typing_done_callback` as it accepts `asyncio.Future` instead of `asyncio.Task`